From 9e5c461efa466397a1a77597c28ac6011bc8ef69 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:56 -0400
Subject: [tools] perf/evsel: Always use perf_evsel__set_sample_bit()

Message-id: <1408462094-14194-248-git-send-email-jolsa@redhat.com>
Patchwork-id: 88229
O-Subject: [PATCH RHEL7.1 BZ1131394 247/325] perf evsel: Always use perf_evsel__set_sample_bit()
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 1e7ed5ec54e3998bda6ea625599a0644404cb421
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Nov 1 15:51:35 2013 +0200

description
===========
Always use perf_evsel__set_sample_bit() rather than just setting the
bit.

[ Cope with 3090ffb "perf: Disable PERF_RECORD_MMAP2 support" ]
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evsel.c b/tools/perf/util/evsel.c
index f0e65de..47bbf03 100644
--- a/tools/perf/util/evsel.c
+++ b/tools/perf/util/evsel.c
@@ -663,7 +663,7 @@ void perf_evsel__config(struct perf_evsel *evsel,
  }
 
  if (opts->sample_address)
-  attr->sample_type |= PERF_SAMPLE_DATA_SRC;
+  perf_evsel__set_sample_bit(evsel, DATA_SRC);
 
  if (opts->no_delay) {
   attr->watermark = 0;
@@ -675,13 +675,13 @@ void perf_evsel__config(struct perf_evsel *evsel,
  }
 
  if (opts->sample_weight)
-  attr->sample_type |= PERF_SAMPLE_WEIGHT;
+  perf_evsel__set_sample_bit(evsel, WEIGHT);
 
  attr->mmap  = track;
  attr->comm  = track;
 
  if (opts->sample_transaction)
-  attr->sample_type |= PERF_SAMPLE_TRANSACTION;
+  perf_evsel__set_sample_bit(evsel, TRANSACTION);
 
  /*
   * XXX see the function comment above
-- 
1.7.1